js/controllers/guided-tour.js

```diff
@@ -12,6 +14,35 @@ export default class extends Controller {
         this.indexValueChanged();
       });
     }
+
+    this.resetAutoplayTimer();
+  }
+
+  disconnect() {
+    if (this.autoplayInterval) window.clearInterval(this.autoplayInterval);
+    if (this.autoplayTimer) window.clearTimeout(this.autoplayTimer);
+  }
+
+  // enter the auto-play mode where we cycle to the end of the tour and then go back to the
+  // intro slide
+  autoplay() {
+    if (this.autoplayInterval) window.clearInterval(this.autoplayInterval);
+
+    this.autoplayInterval = window.setInterval(() => {
+      if (this.indexValue == 0) return this.nextValue && (window.location = this.nextValue);
+      if (this.ended) return this.indexValue = 0;
+      this.indexValue = this.indexValue + 1
+    }, this.constructor.autoplayIntervalTime);
```

Comment:
this is important and took me a bit to puzzle through, so i think it needs at least some comments. i tend not to write things this way because they're harder for me to read (e.g. `return`ing a value that isn't used because you want to collapse an expression and an early `return`, or short-circuiting with `&&`).

i do like that this also addresses the unattended experience: "autoplay" can be between steps in an experience, or between experiences. does this give us everything we need to implement the flow outlined in #74? being able to pick arbitrary "next" experiences for each experience seems like it does, just want to check.

i think this can be adapted without too much work to the oral history controller too, it just only needs the first case on line 32 (if we're at the beginning, go to the next experience).

Reply:
Yep, I think this chews off a big chunk of #74! Once we have oral histories wired up to this, I think it's good for Gary's review (and input on what time intervals make sense here).